auth/auth-store/src/main/java/com/bazaarvoice/emodb/auth/permissions/TablePermissionManager.java

```diff
@@ -73,34 +69,53 @@ public TablePermissionManager(PermissionResolver permissionResolver, DataStore d
     }
 
     @Override
-    public void updateForRole(String role, PermissionUpdateRequest request) {
-        update(getRoleKey(role), request);
-    }
-
-    private void update(String key, PermissionUpdateRequest request) {
+    public void updatePermissions(String id, PermissionUpdateRequest request) {
+        checkNotNull(id, "id");
         checkNotNull(request, "request");
         validateTable();
 
```

Comment:
My original comment was not about validating. It was about checking the removeRest property and act on it here (to clear the existing data before updating new ones), but now when I look the code again, I see that we are considering that flag when we create and build the Delta. It looks good to me now 👍